pkg/model/mask.go

```diff
@@ -115,6 +110,53 @@ func updateField(src, dst reflect.Value, fields []string) {
 	}
 }
 
+// updateSliceField this function updates a destination slice by either replacing a struct with a matching ID
+// or appending new values if no matching ID is found.
+func updateSliceField(srcField, dstField reflect.Value) {
+	appendOnly := func() bool {
+		el := srcField.Type().Elem()
+
+		if el.Kind() != reflect.Struct {
+			return true
+		}
+
+		_, ok := el.FieldByName("ID")
+
+		return !ok
+	}
+
+	if appendOnly() {
+		dstField.Set(reflect.AppendSlice(dstField, srcField))
+		return
+	}
+
+	// If found by id, replace the corresponding structure in the destination slice. Otherwise, append.
+	for i := 0; i < srcField.Len(); i++ {
+		srcID := srcField.Index(i).FieldByName("ID").Interface()
+		idx := findInDstField(srcID, dstField)
+
+		// Check if the "ID" field of the first element in the source slice
+		// matches the "ID" field of the current element in the destination slice.
+		if idx != -1 {
+			// If found, update the corresponding structure in the destination slice
+			dstField.Index(idx).Set(srcField.Index(i))
```

Comment:
The code is self-descriptive. No need for comments.

My personal preference, emphasizes all valid values, inclusive range.

```suggestion
		if idx >= 0 {
			dstField.Index(idx).Set(srcField.Index(i))
```
Emphasizes all valid values, exclusive range.

```suggestion
		if idx > -1 {
			dstField.Index(idx).Set(srcField.Index(i))
```